CalendarApp/GoogleCalendarService/Manager/EventPeriodicityHolder.cs

```diff
@@ -0,0 +1,157 @@
+﻿using System;
+using System.Collections.Generic;
+using Google.Apis.Calendar.v3.Data;
+
+namespace GoogleCalendarService.Manager
+{
+    public class EventPeriodicityHolder
+    {
+        private readonly Event _realEvent;
+        private readonly string _frequency;
+        private List<DayOfWeek> _dayOfWeeksList;
+        private readonly int _count;
+        public List<Event> EventList;
+        private DateTime _startDateTime;
+        private DateTime _endDateTime;
+
+        public EventPeriodicityHolder(Event eventForAdd)
+        {
+            if (eventForAdd.Recurrence == null)
+            {
+                _dayOfWeeksList = new List<DayOfWeek>();
+                _dayOfWeeksList.Add(eventForAdd.Start.DateTime.Value.DayOfWeek);
+                _count = 1;
+                _frequency = "DAILY";
+            }
+            else
+            {
+                string mainRule = eventForAdd.Recurrence[0].Split(':')[1];
+                string[] rules = mainRule.Split(';');
+                foreach (var rule in rules)
+                {
+                    var ruleStr = rule.Split('=');
+                    if (ruleStr[0] == "FREQ")
+                        _frequency = ruleStr[1];
+                    else if (ruleStr[0] == "BYDAY")
+                        DayOfWeeksListSet(ruleStr[1]);
+                    else if (ruleStr[0] == "COUNT")
+                        _count = Convert.ToInt32(ruleStr[1]);
```

Comment:
Or maybe it will be even better to use regular expression to parse the string and get required groups